### PR TITLE
digamma and trigamma accuracy improvements

### PR DESCRIPTION
**Pull request by KlausC**
Solves issue  #450

### CODE REVIEW

**Comment by stevengj** [comment]
LGTM.

**Review by stevengj — APPROVED** [review]

**Comment by KlausC** [comment]
I added the changes for `trigamma`and also consequently replaced `sin(pi*x)` etc. with `sinpi(x)` to obtain exact values at zeros and poles of the functions.

**Comment by KlausC** [comment]
The codecov/patch check seems wrong, because the uncovered lines are not part of the changes made in this PR.

File: src/gamma.jl [hunk 1]
```diff
@@ -56,6 +57,12 @@ function _digamma(x::BigFloat)
     return z
 end
 
+"""
+    _cotpi(x) = cot(π * x)
+
+Accurate for integer arguments
+"""
+_cotpi(x) = cospi(x) / sinpi(x)
```
Comment (devmotion) [inline src/gamma.jl:65]: On Julia 1.10, we can use `tanpi`:
```suggestion
if VERSION >= v"1.10.0-DEV.525"
    _cotpi(x) = inv(tanpi(x))
else
    _cotpi(x) = cospi(x) / sinpi(x)
end
```

Comment (KlausC) [reply src/gamma.jl:65]: will do!

File: src/gamma.jl [hunk 2]
```diff
@@ -132,12 +141,12 @@ const cotderiv_Q = [cotderiv_q(m) for m in 1:100]
 function cotderiv(m::Integer, z)
     isinf(imag(z)) && return zero(z)
     if m <= 0
-        m == 0 && return π * cot(π*z)
+        m == 0 && return π * _cotpi(z)
```
Comment (devmotion) [inline src/gamma.jl:144]: It seems codecov complains about this line (and the following one). Maybe we could add a test for it and move it before the `if` statement? IMO this could be left for a separate PR though.

**Comment by KlausC** [comment]
I am finished with this PR. If there are no concerns, it can be merged now.

**Comment by ViralBShah** [comment]
@KlausC I invited you to the JuliaMath org so that these PRs don't sit around for too long once approved.
